cmake/SetupBoost.cmake

```diff
@@ -120,3 +120,13 @@ set_property(TARGET Boost::boost
   APPEND PROPERTY
   INTERFACE_COMPILE_DEFINITIONS
   $<$<COMPILE_LANGUAGE:CXX>:${BOOST_MULTI_ARRAY_TYPES_HEADER_GUARD}>)
+
+# Work around boost not building with clang 15 (fixed in boost 1.83.0)
+# (https://github.com/boostorg/functional/pull/21)
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND BOOST_VERSION VERSION_LESS 1.83.0)
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "15.0.0" OR
+      CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "15.0.0")
```

Comment:
Note that `GREATER_EQUAL` also exists, and this could also be combined with the above `if` (not requesting this).